lib/ansible/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -893,28 +1073,70 @@ def main():
   except Exception as e:
     module.fail_json(msg='authorization on proxmox cluster failed with exception: %s' % e)
 
-
   # If vmid not set get the Next VM id from ProxmoxAPI
   # If vm name is set get the VM id from ProxmoxAPI
-  if module.params['vmid'] is not None:
-    vmid = module.params['vmid']
-  elif state == 'present':
-    vmid = get_nextvmid(proxmox)
-  elif module.params['name'] is not None:
-    vmid = get_vmid(proxmox, name)[0]
+
+  if module.params['delete'] is not None and module.params['revert'] is not None:
```

Comment:
Their are built in helpers for argspec  for the this, have a look in other modules for "exclusive"